pkgs/applications/editors/neovim/wrapper.nix

```
@@ -193,8 +193,9 @@ let
             "${placeholder "out"}/rplugin.vim"
           ]
           ++ lib.optionals finalAttrs.wrapRc [
-            "--add-flags"
-            "-u ${writeText "init.lua" rcContent}"
+            "--set-default"
+            "VIMINIT"
+            "lua dofile([[${writeText "init.lua" rcContent}]])"
```

It should be safe to assume a nix store path won't contain `'`, `"`, _or_ `]` characters, since NAR hashes won't and the filename doesn't.

Would using normal quotes be more readable?

```suggestion
            "lua dofile('${writeText "init.lua" rcContent}')"
```

If we wanted to be safe, we could use `lib.generators.toLua` to escape the string. Internally, `toLua` uses `builtins.toJSON` to serialise strings.

```suggestion
            "lua dofile(${toLua { } (writeText "init.lua" rcContent)})"
```

Or

```suggestion
            "lua dofile(${toJSON (writeText "init.lua" rcContent)})"
```

This could be useful when a non-default nix store is used, as such a store path could potentially contain unexpected characters.

---

Hmmm

I think we should use `'` instead of `[[`

writeText will never give us a path containing `'` but it might give us one containing `]]

more validation than that is unneccessary.